File: app/src/main/java/uk/co/catedroid/app/data/repo/NotesRepo.java
package uk.co.catedroid.app.data.repo;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.MutableLiveData;
import android.util.Log;

import java.util.List;

import javax.inject.Inject;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;
import uk.co.catedroid.app.api.CateService;
import uk.co.catedroid.app.data.model.Note;

public class NotesRepo {
    private CateService cateService;

    @Inject
    public NotesRepo(CateService cateService) {
        this.cateService = cateService;
    }

    public LiveData<List<Note>> getNotes() {
        final MutableLiveData<List<Note>> data = new MutableLiveData<>();
        cateService.getNotes().enqueue(new Callback<List<Note>>() {
            @Override
            public void onResponse(Call<List<Note>> call, Response<List<Note>> response) {
                data.setValue(response.body());
            }

            @Override
            public void onFailure(Call<List<Note>> call, Throwable t) {
                Log.e("CATe", "NotesRepo - Failed to get notes: " + t.getMessage(), t);
            }
        });

        return data;
    }
}

File: app/src/main/java/uk/co/catedroid/app/data/repo/NotesRepo.kt
package uk.co.catedroid.app.data.repo

import android.arch.lifecycle.LiveData
import android.arch.lifecycle.MutableLiveData
import android.util.Log

import javax.inject.Inject

import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import uk.co.catedroid.app.api.CateService
import uk.co.catedroid.app.data.model.Note

class NotesRepo @Inject
constructor(private val cateService: CateService) {

    val notes: LiveData<List<Note>>
        get() {
            val data = MutableLiveData<List<Note>>()
            cateService.notes.enqueue(object : Callback<List<Note>> {
                override fun onResponse(call: Call<List<Note>>, response: Response<List<Note>>) {
                    data.value = response.body()
                }

                override fun onFailure(call: Call<List<Note>>, t: Throwable) {
                    Log.e("CATe", "NotesRepo - Failed to get notes: " + t.message, t)
                }
            })

            return data
        }
}
